[Bug] JVM target discovery failure when pod restarts

Hi team, 
I am trying to connect TechEmpower Quarkus application with Cryostat to collect JDK Flight recordings.
I have configured the Quarkus application using JMX with below configuration:

JVM flags:
```
-Dcom.sun.management.jmxremote.port=9091
-Dcom.sun.management.jmxremote.ssl=false
-Dcom.sun.management.jmxremote.authenticate=false
```

pod specs:
```yaml
containers:
  - name: tfb-server
    image: 'quay.io/shbirada/tfb-qrh:1.13.2.F_et17'
    imagePullPolicy: IfNotPresent
    env:
      - name: JAVA_OPTIONS
        value: >-
          -server
          -Dcom.sun.management.jmxremote.port=9091
          -Dcom.sun.management.jmxremote.ssl=false
          -Dcom.sun.management.jmxremote.authenticate=false
    ports:
      - containerPort: 8080
```
service specs:
```yaml
 ports:
  - port: 8080
    targetPort: 8080
    name: tfb-qrh-port
  - port: 9091
    targetPort: 9091
    name: jfr-jmx
```

Whenever the pod restarts the Cryostat Client fails to discover the JVM target . I have tried deleting the existing deployment and re-deploying the quarkus yaml file multiple times but nothing seems to be working.

Cryostat pod description:
```
Events:
  Type    Reason          Age   From               Message
  ----    ------          ----  ----               -------
  Normal  Scheduled       81m   default-scheduler  Successfully assigned tfb/cryostat-tfb1-76b5f684bf-r2dz6 to worker-2.tfbcluster.lab.upshift.rdu2.redhat.com
  Normal  AddedInterface  81m   multus             Add eth0 [10.128.3.110/23] from ovn-kubernetes
  Normal  Pulled          81m   kubelet            Container image "registry.redhat.io/cryostat-tech-preview/cryostat-rhel8@sha256:5a58a0f654f321bdfc91bee2dfaffb43bc018a2007dc5123029053e3b09d5d18" already present on machine
  Normal  Created         81m   kubelet            Created container cryostat-tfb1
  Normal  Started         81m   kubelet            Started container cryostat-tfb1
  Normal  Pulled          81m   kubelet            Container image "registry.redhat.io/cryostat-tech-preview/cryostat-grafana-dashboard-rhel8@sha256:c85e96b2a13c44efa6e595ac3ece194b178fc839de5acc6b17375ad8821a5f5d" already present on machine
  Normal  Created         81m   kubelet            Created container cryostat-tfb1-grafana
  Normal  Started         81m   kubelet            Started container cryostat-tfb1-grafana
  Normal  Pulled          81m   kubelet            Container image "registry.redhat.io/cryostat-tech-preview/jfr-datasource-rhel8@sha256:2b6cb93a32fd7290ea6f49da11a54ee6e5d3f50e87ee1ea56eb54419fc3a1671" already present on machine
  Normal  Created         81m   kubelet            Created container cryostat-tfb1-jfr-datasource
  Normal  Started         81m   kubelet            Started container cryostat-tfb1-jfr-datasource
```
Cryostat pod logs Authentication error, but I am able to access the Cryostat Dashboard using the same credentials from Red Hat OpenShift Platform.


```
Defaulted container "cryostat-tfb1" out of: cryostat-tfb1, cryostat-tfb1-grafana, cryostat-tfb1-jfr-datasource
Importing certificate /truststore/operator/..2023_07_05_08_57_56.1003949320/cryostat-tfb1-ca.crt ...
Certificate was added to keystore
[Storing /opt/cryostat.d/truststore.p12]
+ exec java -XX:+CrashOnOutOfMemoryError -Dcom.sun.management.jmxremote.port=9091 -Dcom.sun.management.jmxremote.rmi.port=9091 -Djavax.net.ssl.trustStore=/opt/cryostat.d/truststore.p12 -Djavax.net.ssl.trustStorePassword=t-4tT7JpyeG1MHB9szlQY7I3rNaGZJr9 -Dcom.sun.management.jmxremote.autodiscovery=false -Dcom.sun.management.jmxremote.authenticate=true -Dcom.sun.management.jmxremote.password.file=/tmp/jmxremote.password -Dcom.sun.management.jmxremote.access.file=/tmp/jmxremote.access -Dcom.sun.management.jmxremote.ssl.need.client.auth=true -Djavax.net.ssl.keyStore=/var/run/secrets/operator.cryostat.io/cryostat-tfb1-tls/keystore.p12 -Djavax.net.ssl.keyStorePassword=nXDjdGLUsjHWK9F4jAVl -Dcom.sun.management.jmxremote.ssl=true -Dcom.sun.management.jmxremote.registry.ssl=true -cp '/app/resources:/app/classes:/app/libs/*:/opt/cryostat.d/clientlib.d/*' @/app/jib-main-class-file
Jul 05, 2023 8:57:58 AM io.cryostat.core.log.Logger info
INFO: Local config path set as /opt/cryostat.d/conf.d
Jul 05, 2023 8:57:58 AM org.hibernate.jpa.internal.util.LogHelper logPersistenceUnitInformation
INFO: HHH000204: Processing PersistenceUnitInfo [name: io.cryostat]
Jul 05, 2023 8:57:58 AM org.hibernate.Version logVersion
INFO: HHH000412: Hibernate ORM core version 5.6.14.Final-redhat-00001
Jul 05, 2023 8:57:58 AM org.hibernate.annotations.common.reflection.java.JavaReflectionManager <clinit>
INFO: HCANN000001: Hibernate Commons Annotations {5.1.2.Final-redhat-00005}
Jul 05, 2023 8:57:59 AM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl configure
WARN: HHH10001002: Using Hibernate built-in connection pool (not for production use!)
Jul 05, 2023 8:57:59 AM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl buildCreator
INFO: HHH10001005: using driver [org.h2.Driver] at URL [jdbc:h2:file:/opt/cryostat.d/conf.d/h2;INIT=create domain if not exists jsonb as varchar]
Jul 05, 2023 8:57:59 AM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl buildCreator
INFO: HHH10001001: Connection properties: {password=****, user=cryostat-tfb1}
Jul 05, 2023 8:57:59 AM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl buildCreator
INFO: HHH10001003: Autocommit mode: false
Jul 05, 2023 8:57:59 AM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl$PooledConnections <init>
INFO: HHH000115: Hibernate connection pool size: 20 (min=1)
Jul 05, 2023 8:57:59 AM org.hibernate.dialect.Dialect <init>
INFO: HHH000400: Using dialect: org.hibernate.dialect.H2Dialect
Jul 05, 2023 8:57:59 AM org.hibernate.resource.transaction.backend.jdbc.internal.DdlTransactionIsolatorNonJtaImpl getIsolatedConnection
INFO: HHH10001501: Connection obtained from JdbcConnectionAccess [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator$ConnectionProviderJdbcConnectionAccess@63d2b4a6] for (non-JTA) DDL execution was not in auto-commit mode; the Connection 'local transaction' will be committed and the Connection will be set into auto-commit mode.
Jul 05, 2023 8:57:59 AM org.hibernate.engine.transaction.jta.platform.internal.JtaPlatformInitiator initiateService
INFO: HHH000490: Using JtaPlatform implementation: [org.hibernate.engine.transaction.jta.platform.internal.NoJtaPlatform]
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: cryostat started, version: 2.3.0.redhat-00004-0-g47f9627.
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Selected SSL KeyStore strategy with keystore /var/run/secrets/operator.cryostat.io/cryostat-tfb1-tls/keystore.p12
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.net.HttpServer Verticle with options: {
  "ha" : false,
  "instances" : 1,
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "worker" : false,
  "workerPoolSize" : 20
}
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: HTTPS service running on https://cryostat-tfb1-tfb.apps.tfbcluster.lab.upshift.rdu2.redhat.com:443
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.net.HttpServer Verticle [e159c5aa-785c-42bf-b4f2-2f39feb86806]
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Selecting configured AuthManager "io.cryostat.net.openshift.OpenShiftAuthManager"
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Local save path for flight recordings set as /opt/cryostat.d/recordings.d
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.net.web.WebServer Verticle with options: {
  "ha" : false,
  "instances" : 1,
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "worker" : false,
  "workerPoolSize" : 20
}
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.net.web.WebServer Verticle [470d4349-5276-430e-b7f9-540a35658339]
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.messaging.MessagingServer Verticle with options: {
  "ha" : false,
  "instances" : 1,
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "worker" : false,
  "workerPoolSize" : 20
}
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Max concurrent WebSocket connections: 2147483647
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.messaging.MessagingServer Verticle [fb282490-038c-4865-aa1c-27a3d576db35]
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.rules.RuleProcessor Verticle with options: {
  "ha" : false,
  "instances" : 1,
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "worker" : true,
  "workerPoolName" : "io.cryostat.rules.RuleProcessor-worker",
  "workerPoolSize" : 20
}
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.rules.RuleProcessor Verticle [a9d5202f-ae4e-4f2b-8781-e7f29d2dc29c]
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.recordings.RecordingMetadataManager Verticle with options: {
  "ha" : false,
  "instances" : 1,
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "worker" : true,
  "workerPoolName" : "io.cryostat.recordings.RecordingMetadataManager-worker",
  "workerPoolSize" : 20
}
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Peeking contents of metadata directory: lost
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Skipping metadata validation: appears to be a special location: lost
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Peeking contents of metadata directory: I5KFG33XNFAWGNSOL5YFK4LNPJBFSTTPGJZVO2BSM5EUCMBSNUZWOWSRKEYWSZSOGNRXOPI=
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deleting empty recording metadata directory: /opt/cryostat.d/conf.d/metadata/I5KFG33XNFAWGNSOL5YFK4LNPJBFSTTPGJZVO2BSM5EUCMBSNUZWOWSRKEYWSZSOGNRXOPI=
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.recordings.RecordingMetadataManager Verticle [beb23067-760a-419e-b904-c74989a86106]
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.discovery.DiscoveryStorage Verticle with options: {
  "ha" : false,
  "instances" : 1,
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "worker" : true,
  "workerPoolName" : "io.cryostat.discovery.DiscoveryStorage-worker",
  "workerPoolSize" : 20
}
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.discovery.BuiltInDiscovery Verticle with options: {
  "ha" : false,
  "instances" : 1,
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "worker" : true,
  "workerPoolName" : "io.cryostat.discovery.BuiltInDiscovery-worker",
  "workerPoolSize" : 20
}
Jul 05, 2023 8:58:00 AM io.cryostat.core.log.Logger info
INFO: Selected KubeApiPlatformStrategy Strategy
Jul 05, 2023 8:58:01 AM io.cryostat.core.log.Logger info
INFO: Started Endpoints SharedInformer for namespace "tfb"
Jul 05, 2023 8:58:01 AM io.cryostat.core.log.Logger info
INFO: Selected PodmanPlatformStrategy Strategy
Jul 05, 2023 8:58:01 AM io.cryostat.core.log.Logger info
INFO: Selected Default Platform Strategy
Jul 05, 2023 8:58:01 AM io.cryostat.core.log.Logger info
INFO: Starting built-in discovery with CustomTargetPlatformClient
Jul 05, 2023 8:58:01 AM io.cryostat.core.log.Logger info
INFO: Selected OpenShiftPlatformStrategy Strategy
Jul 05, 2023 8:58:01 AM io.cryostat.core.log.Logger info
INFO: Starting built-in discovery with KubeApiPlatformClient
Jul 05, 2023 8:58:02 AM io.cryostat.core.log.Logger info
INFO: Started Endpoints SharedInformer for namespace "tfb"
Jul 05, 2023 8:58:02 AM io.cryostat.core.log.Logger info
INFO: Observing new target: io.cryostat.platform.ServiceRef@6fc52ffd[alias=tfb-qrh-sample-7bc8684948-frjwz,annotations=io.cryostat.platform.ServiceRef$Annotations@7eb699c1[cryostat={HOST=10.128.3.97, PORT=9091, NAMESPACE=tfb, POD_NAME=tfb-qrh-sample-7bc8684948-frjwz, REALM=KubernetesApi},platform={openshift.io/scc=restricted-v2, seccomp.security.alpha.kubernetes.io/pod=runtime/default, k8s.ovn.org/pod-networks={"default":{"ip_addresses":["10.128.3.97/23"],"mac_address":"0a:58:0a:80:03:61","gateway_ips":["10.128.2.1"],"ip_address":"10.128.3.97/23","gateway_ip":"10.128.2.1"}}, k8s.v1.cni.cncf.io/network-status=[{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.128.3.97"
    ],
    "mac": "0a:58:0a:80:03:61",
    "default": true,
    "dns": {}
}]}],jvmId=<null>,labels={app=tfb-qrh-deployment, app.kubernetes.io/name=tfb-qrh-deployment, name=tfb-qrh-deployment, pod-template-hash=7bc8684948, version=v1, app.kubernetes.io/layer=quarkus},serviceUri=service:jmx:rmi:///jndi/rmi://10-128-3-97.tfb.pod:9091/jmxrmi]
Jul 05, 2023 8:58:02 AM io.cryostat.core.log.Logger info
INFO: Creating connection for service:jmx:rmi:///jndi/rmi://10-128-3-97.tfb.pod:9091/jmxrmi
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.discovery.BuiltInDiscovery Verticle [1315954f-b235-4b27-a4e8-3a1b90a2130e]
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.discovery.DiscoveryStorage Verticle [f1f85698-8dc7-45e6-af87-c6904cd98690]
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Outgoing WS message: {"meta":{"category":"TargetJvmDiscovery","type":{"type":"application","subType":"json"},"serverTime":1688547481},"message":{"event":{"serviceRef":{"jvmId":null,"connectUrl":"service:jmx:rmi:///jndi/rmi://10-131-0-25.tfb.pod:9091/jmxrmi","alias":"cryostat-tfb1-76b5f684bf-m2chk","labels":{"app":"cryostat-tfb1","pod-template-hash":"76b5f684bf","component":"cryostat","kind":"cryostat"},"annotations":{"platform":{"seccomp.security.alpha.kubernetes.io/pod":"runtime/default","openshift.io/scc":"restricted-v2","k8s.ovn.org/pod-networks":"{\"default\":{\"ip_addresses\":[\"10.131.0.25/23\"],\"mac_address\":\"0a:58:0a:83:00:19\",\"gateway_ips\":[\"10.131.0.1\"],\"ip_address\":\"10.131.0.25/23\",\"gateway_ip\":\"10.131.0.1\"}}","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"ovn-kubernetes\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.131.0.25\"\n    ],\n    \"mac\": \"0a:58:0a:83:00:19\",\n    \"default\": true,\n    \"dns\": {}\n}]"},"cryostat":{"HOST":"10.131.0.25","PORT":"9091","NAMESPACE":"tfb","POD_NAME":"cryostat-tfb1-76b5f684bf-m2chk","REALM":"KubernetesApi"}}},"kind":"LOST"}}}
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Outgoing WS message: {"meta":{"category":"TargetJvmDiscovery","type":{"type":"application","subType":"json"},"serverTime":1688547484},"message":{"event":{"serviceRef":{"jvmId":"Ae7TI2hyS81CE8zxQ4SU_mivHizT18XpON68sV1wwZQ=","connectUrl":"service:jmx:rmi:///jndi/rmi://10-128-3-97.tfb.pod:9091/jmxrmi","alias":"tfb-qrh-sample-7bc8684948-frjwz","labels":{"app":"tfb-qrh-deployment","app.kubernetes.io/name":"tfb-qrh-deployment","name":"tfb-qrh-deployment","pod-template-hash":"7bc8684948","version":"v1","app.kubernetes.io/layer":"quarkus"},"annotations":{"platform":{"seccomp.security.alpha.kubernetes.io/pod":"runtime/default","openshift.io/scc":"restricted-v2","k8s.ovn.org/pod-networks":"{\"default\":{\"ip_addresses\":[\"10.128.3.97/23\"],\"mac_address\":\"0a:58:0a:80:03:61\",\"gateway_ips\":[\"10.128.2.1\"],\"ip_address\":\"10.128.3.97/23\",\"gateway_ip\":\"10.128.2.1\"}}","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"ovn-kubernetes\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.128.3.97\"\n    ],\n    \"mac\": \"0a:58:0a:80:03:61\",\n    \"default\": true,\n    \"dns\": {}\n}]"},"cryostat":{"HOST":"10.128.3.97","PORT":"9091","NAMESPACE":"tfb","POD_NAME":"tfb-qrh-sample-7bc8684948-frjwz","REALM":"KubernetesApi"}}},"kind":"FOUND"}}}
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Starting archive migration
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Skipping archive migration: appears to be a special location: file-uploads
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Skipping archive migration: appears to be a special location: lost
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Successfully migrated archives
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Beginning to prune potentially stale metadata...
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: Successfully pruned all stale metadata
Jul 05, 2023 8:58:04 AM io.cryostat.core.log.Logger info
INFO: JVM ID: service:jmx:rmi:///jndi/rmi://10-128-3-97.tfb.pod:9091/jmxrmi -> Ae7TI2hyS81CE8zxQ4SU_mivHizT18XpON68sV1wwZQ=
Jul 05, 2023 8:58:06 AM org.slf4j.impl.JDK14LoggerAdapter fillCallerData
INFO: 10.128.2.2 - - [Wed, 5 Jul 2023 08:58:06 GMT] 3ms "GET /health/liveness HTTP/1.1" 204 0 bytes "-" "kube-probe/1.26"
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Observing new target: io.cryostat.platform.ServiceRef@7f6feb45[alias=cryostat-tfb1-76b5f684bf-r2dz6,annotations=io.cryostat.platform.ServiceRef$Annotations@71a80ad8[cryostat={HOST=10.128.3.110, PORT=9091, NAMESPACE=tfb, POD_NAME=cryostat-tfb1-76b5f684bf-r2dz6, REALM=KubernetesApi},platform={openshift.io/scc=restricted-v2, seccomp.security.alpha.kubernetes.io/pod=runtime/default, k8s.ovn.org/pod-networks={"default":{"ip_addresses":["10.128.3.110/23"],"mac_address":"0a:58:0a:80:03:6e","gateway_ips":["10.128.2.1"],"ip_address":"10.128.3.110/23","gateway_ip":"10.128.2.1"}}, k8s.v1.cni.cncf.io/network-status=[{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.128.3.110"
    ],
    "mac": "0a:58:0a:80:03:6e",
    "default": true,
    "dns": {}
}]}],jvmId=<null>,labels={app=cryostat-tfb1, pod-template-hash=76b5f684bf, component=cryostat, kind=cryostat},serviceUri=service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi]
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Creating connection for service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: connection attempt failed.
 
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Connection for service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi closed
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger error
SEVERE: Exception thrown
org.openjdk.jmc.rjmx.ConnectionException caused by java.lang.SecurityException: Authentication failed! Invalid username or password
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:345)
        at io.cryostat.core.net.JFRJMXConnection.attemptConnect(JFRJMXConnection.java:443)
        at io.cryostat.core.net.JFRJMXConnection.connect(JFRJMXConnection.java:399)
        at io.cryostat.core.net.JFRJMXConnection.getJvmId(JFRJMXConnection.java:204)
        at io.cryostat.net.TargetConnectionManager.lambda$executeConnectedTaskAsync$2(TargetConnectionManager.java:153)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
        at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.SecurityException: Authentication failed! Invalid username or password
        at java.management/com.sun.jmx.remote.security.JMXPluggableAuthenticator.authenticationFailure(JMXPluggableAuthenticator.java:231)
        at java.management/com.sun.jmx.remote.security.JMXPluggableAuthenticator.authenticate(JMXPluggableAuthenticator.java:208)
        at jdk.management.agent/sun.management.jmxremote.ConnectorBootstrap$AccessFileCheckerAuthenticator.authenticate(ConnectorBootstrap.java:233)
        at java.management.rmi/javax.management.remote.rmi.RMIServerImpl.doNewClient(RMIServerImpl.java:231)
        at java.management.rmi/javax.management.remote.rmi.RMIServerImpl.newClient(RMIServerImpl.java:198)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at java.rmi/sun.rmi.transport.StreamRemoteCall.exceptionReceivedFromServer(StreamRemoteCall.java:304)
        at java.rmi/sun.rmi.transport.StreamRemoteCall.executeCall(StreamRemoteCall.java:280)
        at java.rmi/sun.rmi.server.UnicastRef.invoke(UnicastRef.java:165)
        at java.management.rmi/javax.management.remote.rmi.RMIServerImpl_Stub.newClient(RMIServerImpl_Stub.java:83)
        at java.management.rmi/javax.management.remote.rmi.RMIConnector.getConnection(RMIConnector.java:2107)
        at java.management.rmi/javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:321)
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.connectJmxConnector(RJMXConnection.java:575)
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.establishConnection(RJMXConnection.java:560)
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:338)
        ... 11 more
Caused by: javax.security.auth.login.FailedLoginException: Invalid username or password
        at java.management/com.sun.jmx.remote.security.FileLoginModule.attemptAuthentication(FileLoginModule.java:460)
        at java.management/com.sun.jmx.remote.security.FileLoginModule.login(FileLoginModule.java:309)
        at java.base/javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
        at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:679)
        at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:677)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.base/javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:677)
        at java.base/javax.security.auth.login.LoginContext.login(LoginContext.java:587)
        at java.management/com.sun.jmx.remote.security.JMXPluggableAuthenticator.authenticate(JMXPluggableAuthenticator.java:195)
        at jdk.management.agent/sun.management.jmxremote.ConnectorBootstrap$AccessFileCheckerAuthenticator.authenticate(ConnectorBootstrap.java:233)
        at java.management.rmi/javax.management.remote.rmi.RMIServerImpl.doNewClient(RMIServerImpl.java:231)
        at java.management.rmi/javax.management.remote.rmi.RMIServerImpl.newClient(RMIServerImpl.java:198)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
 
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Removing cached connection for service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi: EXPLICIT
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger warn
WARNING: Could not resolve jvmId for target service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Update node [service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi] with null jvmId
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Observing new target: io.cryostat.platform.ServiceRef@36c74ad2[alias=tfb-qrh-sample-7bc8684948-frjwz,annotations=io.cryostat.platform.ServiceRef$Annotations@410b26b1[cryostat={HOST=10.128.3.97, PORT=9091, NAMESPACE=tfb, POD_NAME=tfb-qrh-sample-7bc8684948-frjwz, REALM=KubernetesApi},platform={openshift.io/scc=restricted-v2, seccomp.security.alpha.kubernetes.io/pod=runtime/default, k8s.ovn.org/pod-networks={"default":{"ip_addresses":["10.128.3.97/23"],"mac_address":"0a:58:0a:80:03:61","gateway_ips":["10.128.2.1"],"ip_address":"10.128.3.97/23","gateway_ip":"10.128.2.1"}}, k8s.v1.cni.cncf.io/network-status=[{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.128.3.97"
    ],
    "mac": "0a:58:0a:80:03:61",
    "default": true,
    "dns": {}
}]}],jvmId=<null>,labels={app=tfb-qrh-deployment, app.kubernetes.io/name=tfb-qrh-deployment, name=tfb-qrh-deployment, pod-template-hash=7bc8684948, version=v1, app.kubernetes.io/layer=quarkus},serviceUri=service:jmx:rmi:///jndi/rmi://10-128-3-97.tfb.pod:9091/jmxrmi]
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Outgoing WS message: {"meta":{"category":"TargetJvmDiscovery","type":{"type":"application","subType":"json"},"serverTime":1688547487},"message":{"event":{"serviceRef":{"jvmId":null,"connectUrl":"service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi","alias":"cryostat-tfb1-76b5f684bf-r2dz6","labels":{"app":"cryostat-tfb1","pod-template-hash":"76b5f684bf","component":"cryostat","kind":"cryostat"},"annotations":{"platform":{"seccomp.security.alpha.kubernetes.io/pod":"runtime/default","openshift.io/scc":"restricted-v2","k8s.ovn.org/pod-networks":"{\"default\":{\"ip_addresses\":[\"10.128.3.110/23\"],\"mac_address\":\"0a:58:0a:80:03:6e\",\"gateway_ips\":[\"10.128.2.1\"],\"ip_address\":\"10.128.3.110/23\",\"gateway_ip\":\"10.128.2.1\"}}","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"ovn-kubernetes\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.128.3.110\"\n    ],\n    \"mac\": \"0a:58:0a:80:03:6e\",\n    \"default\": true,\n    \"dns\": {}\n}]"},"cryostat":{"HOST":"10.128.3.110","PORT":"9091","NAMESPACE":"tfb","POD_NAME":"cryostat-tfb1-76b5f684bf-r2dz6","REALM":"KubernetesApi"}}},"kind":"FOUND"}}}
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: Creating connection for service:jmx:rmi:///jndi/rmi://10-128-3-110.tfb.pod:9091/jmxrmi
Jul 05, 2023 8:58:07 AM io.cryostat.core.log.Logger info
INFO: connection attempt failed.
INFO: Outgoing WS message: {"meta":{"category":"WsClientActivity","type":{"type":"application","subType":"json"},"serverTime":1688547806},"message":{"10.128.2.2:35666":"accepted"}}
Jul 05, 2023 9:03:27 AM org.slf4j.impl.JDK14LoggerAdapter fillCallerData
INFO: 10.128.2.2 - - [Wed, 5 Jul 2023 09:03:27 GMT] 79ms "GET /api/v1/recordings HTTP/1.1" 200 9 KB "https://cryostat-tfb1-tfb.apps.tfbcluster.lab.upshift.rdu2.redhat.com/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
Jul 05, 2023 9:03:36 AM org.slf4j.impl.JDK14LoggerAdapter fillCallerData
INFO: 10.128.2.2 - - [Wed, 5 Jul 2023 09:03:36 GMT] 0ms "GET /health/liveness HTTP/1.1" 204 0 bytes "-" "kube-probe/1.26"
Jul 05, 2023 9:03:46 AM org.slf4j.impl.JDK14LoggerAdapter fillCallerData
INFO: 10.128.2.2 - - [Wed, 5 Jul 2023 09:03:46 GMT] 0ms "GET /health/liveness HTTP/1.1" 204 0 bytes "-" "kube-probe/1.26"
Jul 05, 2023 9:03:56 AM org.slf4j.impl.JDK14LoggerAdapter fillCallerData
INFO: 10.128.2.2 - - [Wed, 5 Jul 2023 09:03:56 GMT] 0ms "GET /health/liveness HTTP/1.1" 204 0 bytes "-" "kube-probe/1.26"

```   

Cryostat target is detected but JVM target option cannot be seen
![Cryostat-target](https://github.com/cryostatio/cryostat/assets/70368565/b5ac768a-d568-493d-86ac-1d5182e7aca5)

Topology dashboard:
![topology](https://github.com/cryostatio/cryostat/assets/70368565/e1a5a3f5-ce99-458c-80a3-91db5002a5c3)



## Environment:
- OS: RHEL 8.8
- Environment: OpenShift 4.13.1
- Version: 2.3.0
- Java: OpenJDK version "17.0.6"

Thanks to @maxcao13 , here is a log snippet from a confirmed case where the target discovery failed due to the "JMX not ready" timing issue I was describing:

```
Jul 25, 2023 4:36:20 PM io.cryostat.core.log.Logger info
INFO: Connection for service:jmx:rmi:///jndi/rmi://10-132-0-57.ns1.pod:9097/jmxrmi closed
Jul 25, 2023 4:36:20 PM io.cryostat.core.log.Logger error
SEVERE: Exception thrown
org.openjdk.jmc.rjmx.ConnectionException caused by java.io.IOException: Failed to retrieve RMIServer stub: javax.naming.ServiceUnavailableException [Root exception is java.rmi.ConnectException: Connection refused to host: 10-132-0-57.ns1.pod; nested exception is:
        java.net.ConnectException: Connection refused]
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:345)
        at io.cryostat.core.net.JFRJMXConnection.attemptConnect(JFRJMXConnection.java:443)
        at io.cryostat.core.net.JFRJMXConnection.connect(JFRJMXConnection.java:399)
        at io.cryostat.core.net.JFRJMXConnection.getJvmId(JFRJMXConnection.java:204)
        at io.cryostat.net.TargetConnectionManager.lambda$executeConnectedTaskAsync$2(TargetConnectionManager.java:153)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
        at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.io.IOException: Failed to retrieve RMIServer stub: javax.naming.ServiceUnavailableException [Root exception is java.rmi.ConnectException: Connection refused to host: 10-132-0-57.ns1.pod; nested exception is:
        java.net.ConnectException: Connection refused]
        at java.management.rmi/javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:370)
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.connectJmxConnector(RJMXConnection.java:575)
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.establishConnection(RJMXConnection.java:552)
        at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:338)
        ... 11 more
Caused by: javax.naming.ServiceUnavailableException [Root exception is java.rmi.ConnectException: Connection refused to host: 10-132-0-57.ns1.pod; nested exception is:
        java.net.ConnectException: Connection refused]
        at jdk.naming.rmi/com.sun.jndi.rmi.registry.RegistryContext.lookup(RegistryContext.java:138)
        at java.naming/com.sun.jndi.toolkit.url.GenericURLContext.lookup(GenericURLContext.java:220)
        at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
        at java.management.rmi/javax.management.remote.rmi.RMIConnector.findRMIServerJNDI(RMIConnector.java:1839)
        at java.management.rmi/javax.management.remote.rmi.RMIConnector.findRMIServer(RMIConnector.java:1813)
        at java.management.rmi/javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:302)
        ... 14 more
Caused by: java.rmi.ConnectException: Connection refused to host: 10-132-0-57.ns1.pod; nested exception is:
        java.net.ConnectException: Connection refused
        at java.rmi/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:626)
        at java.rmi/sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:217)
        at java.rmi/sun.rmi.transport.tcp.TCPChannel.newConnection(TCPChannel.java:204)
        at java.rmi/sun.rmi.server.UnicastRef.newCall(UnicastRef.java:344)
        at java.rmi/sun.rmi.registry.RegistryImpl_Stub.lookup(RegistryImpl_Stub.java:116)
        at jdk.naming.rmi/com.sun.jndi.rmi.registry.RegistryContext.lookup(RegistryContext.java:134)
        ... 19 more
Caused by: java.net.ConnectException: Connection refused
        at java.base/sun.nio.ch.Net.connect0(Native Method)
        at java.base/sun.nio.ch.Net.connect(Net.java:579)
        at java.base/sun.nio.ch.Net.connect(Net.java:568)
        at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:588)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
        at java.base/java.net.Socket.connect(Socket.java:633)
        at java.base/java.net.Socket.connect(Socket.java:583)
        at java.base/java.net.Socket.<init>(Socket.java:507)
        at java.base/java.net.Socket.<init>(Socket.java:287)
        at java.rmi/sun.rmi.transport.tcp.TCPDirectSocketFactory.createSocket(TCPDirectSocketFactory.java:40)
        at java.rmi/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:620)
        ... 24 more
 
Jul 25, 2023 4:36:20 PM io.cryostat.core.log.Logger info
INFO: Removing cached connection for service:jmx:rmi:///jndi/rmi://10-132-0-57.ns1.pod:9097/jmxrmi: EXPLICIT
Jul 25, 2023 4:36:20 PM io.cryostat.core.log.Logger warn
WARNING: Could not resolve jvmId for target service:jmx:rmi:///jndi/rmi://10-132-0-57.ns1.pod:9097/jmxrmi
Jul 25, 2023 4:36:20 PM io.cryostat.core.log.Logger info
INFO: Ignoring target node [service:jmx:rmi:///jndi/rmi://10-132-0-57.ns1.pod:9097/jmxrmi]
```

This is the kind of case that some retry logic would address, but it would not help much in the case that you're looking for startup profiling unless the retries are very, very frequent.

--- a/compose/compose-cryostat.yaml
+++ b/compose/compose-cryostat.yaml
@@ -6,7 +6,7 @@ services:
     user: '0'
     stdin_open: true
     tty: true
-    labels: 
+    labels:
       - io.cryostat.discovery=true
       - io.cryostat.jmxUrl=service:jmx:rmi:///jndi/rmi://localhost:0/jmxrmi
     deploy:
@@ -82,7 +82,7 @@ services:
   cryostat-duplicate:
     image: quay.io/cryostat/cryostat:latest
     container_name: cryostat-duplicate
-    labels: 
+    labels:
       - io.cryostat.discovery=true
       - io.cryostat.jmxUrl=service:jmx:rmi:///jndi/rmi://cryostat-duplicate:8282/jmxrmi
       - io.cryostat.jmxPort=8282
@@ -154,9 +154,9 @@ services:
 
   # Testing apps
   scaled-app:
-    image: quay.io/andrewazores/vertx-fib-demo:0.12.2
+    image: quay.io/andrewazores/vertx-fib-demo:0.13.0
     container_name: scaled-app
-    labels: 
+    labels:
       - io.cryostat.jmxUrl=service:jmx:rmi:///jndi/rmi://scaled-app:9093/jmxrmi
     environment:
       - JMX_PORT=9093
@@ -167,9 +167,9 @@ services:
     profiles: ["scaled"]
 
   stopped-app:
-    image: quay.io/andrewazores/vertx-fib-demo:0.12.2
+    image: quay.io/andrewazores/vertx-fib-demo:0.13.0
     container_name: stopped-app
-    labels: 
+    labels:
       - io.cryostat.jmxUrl=service:jmx:rmi:///jndi/rmi://stopped-app:9093/jmxrmi
     environment:
       - JMX_PORT=9093
@@ -181,9 +181,9 @@ services:
 
 # invalid targets:
   invalid-podman:
-    image: quay.io/andrewazores/vertx-fib-demo:0.12.2
+    image: quay.io/andrewazores/vertx-fib-demo:0.13.0
     container_name: invalid-jmx
-    labels: 
+    labels:
       - io.cryostat.jmxUrl=invalid:jmx:rmi:///jndi/rmi://invalid-podman:9093/jmxrmi # invalid serviceurl
     environment:
       - JMX_PORT=9093

--- a/compose/compose-vertx-jmx.yaml
+++ b/compose/compose-vertx-jmx.yaml
@@ -1,7 +1,7 @@
 ---
 services:
   vertx-jmx:
-    image: quay.io/andrewazores/vertx-fib-demo:0.12.4
+    image: quay.io/andrewazores/vertx-fib-demo:0.13.0
     container_name: vertx-jmx
     deploy:
       mode: global

--- a/smoketest-docker.sh
+++ b/smoketest-docker.sh
@@ -106,7 +106,7 @@ runDemoApps() {
         --label io.cryostat.jmxPort="9093" \
         --publish 8081:8081 \
         --publish 9093:9093 \
-        --rm -d quay.io/andrewazores/vertx-fib-demo:0.12.2
+        --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
     CONTAINERS="${CONTAINERS:+${CONTAINERS} }vertx-fib-demo-1"
 
     docker run \
@@ -119,7 +119,7 @@ runDemoApps() {
         --label io.cryostat.jmxPort="9094" \
         --publish 8082:8082 \
         --publish 9094:9092 \
-        --rm -d quay.io/andrewazores/vertx-fib-demo:0.12.2
+        --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
     CONTAINERS="${CONTAINERS:+${CONTAINERS} }vertx-fib-demo-2"
 
     docker run \
@@ -133,7 +133,7 @@ runDemoApps() {
         --label io.cryostat.jmxPort="9095" \
         --publish 8083:8083 \
         --publish 9095:9095 \
-        --rm -d quay.io/andrewazores/vertx-fib-demo:0.12.2
+        --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
     CONTAINERS="${CONTAINERS:+${CONTAINERS} }vertx-fib-demo-3"
 
     # this config is broken on purpose (missing required env vars) to test the agent's behaviour

--- a/smoketest.sh
+++ b/smoketest.sh
@@ -133,7 +133,7 @@ runDemoApps() {
         --label io.cryostat.discovery="true" \
         --label io.cryostat.jmxHost="localhost" \
         --label io.cryostat.jmxPort="9093" \
-        --rm -d quay.io/andrewazores/vertx-fib-demo:0.12.3
+        --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
 
     podman run \
         --name vertx-fib-demo-2 \
@@ -154,7 +154,7 @@ runDemoApps() {
         --label io.cryostat.jmxHost="localhost" \
         --label io.cryostat.jmxPort="9094" \
         --label io.cryostat.jmxUrl="service:jmx:rmi:///jndi/rmi://localhost:9094/jmxrmi" \
-        --rm -d quay.io/andrewazores/vertx-fib-demo:0.12.3
+        --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
 
     podman run \
         --name vertx-fib-demo-3 \
@@ -174,7 +174,7 @@ runDemoApps() {
         --pod cryostat-pod \
         --label io.cryostat.discovery="true" \
         --label io.cryostat.jmxUrl="service:jmx:rmi:///jndi/rmi://localhost:9095/jmxrmi" \
-        --rm -d quay.io/andrewazores/vertx-fib-demo:0.12.3
+        --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
 
     # this config is broken on purpose (missing required env vars) to test the agent's behaviour
     # when not properly set up

--- a/smoketest.sh
+++ b/smoketest.sh
@@ -117,6 +117,17 @@ runDemoApps() {
         --rm -d quay.io/roberttoyonaga/jmx:jmxquarkus@sha256:b067f29faa91312d20d43c55d194a2e076de7d0d094da3d43ee7d2b2b5a6f100
 
     podman run \
+        --name vertx-fib-demo-0 \
+        --env HTTP_PORT=8079 \
+        --env JMX_PORT=9089 \
+        --env START_DELAY=60 \
+        --pod cryostat-pod \
+        --label io.cryostat.discovery="true" \
+        --label io.cryostat.jmxHost="localhost" \
+        --label io.cryostat.jmxPort="9093" \
+        --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
+
+    podman run \
         --name vertx-fib-demo-1 \
         --env HTTP_PORT=8081 \
         --env JMX_PORT=9093 \

--- a/smoketest.sh
+++ b/smoketest.sh
@@ -124,7 +124,7 @@ runDemoApps() {
         --pod cryostat-pod \
         --label io.cryostat.discovery="true" \
         --label io.cryostat.jmxHost="localhost" \
-        --label io.cryostat.jmxPort="9093" \
+        --label io.cryostat.jmxPort="9089" \
         --rm -d quay.io/andrewazores/vertx-fib-demo:0.13.0
 
     podman run \

--- a/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
+++ b/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
@@ -49,6 +49,7 @@ import java.util.Objects;
 import java.util.Optional;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ForkJoinPool;
 
 import javax.script.ScriptException;
 
@@ -65,6 +66,7 @@ import io.cryostat.platform.discovery.BaseNodeType;
 import io.cryostat.platform.discovery.EnvironmentNode;
 import io.cryostat.platform.discovery.TargetNode;
 import io.cryostat.recordings.JvmIdHelper;
+import io.cryostat.recordings.JvmIdHelper.JvmIdGetException;
 import io.cryostat.rules.MatchExpressionEvaluator;
 import io.cryostat.util.HttpStatusCodeIdentifier;
 
@@ -96,9 +98,10 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
     private final Gson gson;
     private final WebClient http;
     private final Logger logger;
-    private long timerId = -1L;
+    private long pluginPruneTimerId = -1L;
+    private long targetRetryTimeId = -1L;
 
-    private final Map<TargetNode, UUID> targetsToUpdate = new HashMap<>();
+    private final Map<TargetNode, UUID> nonConnectableTargets = new HashMap<>();
 
     public static final String DISCOVERY_STARTUP_ADDRESS = "discovery-startup";
 
@@ -143,14 +146,61 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                                                                                 + " deployed")))
                 .onFailure(future::fail);
 
-        this.timerId = getVertx().setPeriodic(pingPeriod.toMillis(), i -> pingPrune());
+        this.pluginPruneTimerId = getVertx().setPeriodic(pingPeriod.toMillis(), i -> pingPrune());
+        this.targetRetryTimeId =
+                getVertx()
+                        .setPeriodic(
+                                15_000,
+                                i -> {
+                                    ForkJoinPool.commonPool()
+                                            .execute(
+                                                    () -> {
+                                                        Map<TargetNode, UUID> copy =
+                                                                new HashMap<>(
+                                                                        nonConnectableTargets);
+                                                        for (var entry : copy.entrySet()) {
+                                                            TargetNode targetNode = entry.getKey();
+                                                            try {
+                                                                ServiceRef sr =
+                                                                        jvmIdHelper
+                                                                                .get()
+                                                                                .resolveId(
+                                                                                        targetNode
+                                                                                                .getTarget());
+                                                                if (StringUtils.isBlank(
+                                                                        sr.getJvmId())) {
+                                                                    continue;
+                                                                }
+                                                                nonConnectableTargets.remove(
+                                                                        targetNode);
+                                                                UUID id = entry.getValue();
+                                                                PluginInfo plugin =
+                                                                        getById(id).orElseThrow();
+                                                                EnvironmentNode original =
+                                                                        gson.fromJson(
+                                                                                plugin.getSubtree(),
+                                                                                EnvironmentNode
+                                                                                        .class);
+                                                                update(id, original.getChildren());
+                                                            } catch (JvmIdGetException e) {
+                                                                logger.info(
+                                                                        "Retain null jvmId for node"
+                                                                                + " [{}]",
+                                                                        targetNode);
+                                                            } catch (JsonSyntaxException e) {
+                                                                throw new RuntimeException(e);
+                                                            }
+                                                        }
+                                                    });
+                                });
         this.credentialsManager
                 .get()
                 .addListener(
                         event -> {
                             switch (event.getEventType()) {
                                 case ADDED:
-                                    Map<TargetNode, UUID> copy = new HashMap<>(targetsToUpdate);
+                                    Map<TargetNode, UUID> copy =
+                                            new HashMap<>(nonConnectableTargets);
                                     for (var entry : copy.entrySet()) {
                                         try {
                                             if (matchExpressionEvaluator
@@ -158,7 +208,7 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                                                     .applies(
                                                             event.getPayload(),
                                                             entry.getKey().getTarget())) {
-                                                targetsToUpdate.remove(entry.getKey());
+                                                nonConnectableTargets.remove(entry.getKey());
                                                 UUID id = entry.getValue();
                                                 PluginInfo plugin = getById(id).orElseThrow();
                                                 EnvironmentNode original =
@@ -183,7 +233,8 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
 
     @Override
     public void stop() {
-        getVertx().cancelTimer(timerId);
+        getVertx().cancelTimer(pluginPruneTimerId);
+        getVertx().cancelTimer(targetRetryTimeId);
     }
 
     private CompositeFuture pingPrune() {
@@ -316,6 +367,7 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                 try {
                     ref = jvmIdHelper.get().resolveId(ref);
                 } catch (Exception e) {
+                    nonConnectableTargets.putIfAbsent((TargetNode) child, id);
                     // if Exception is of SSL or JMX Auth, ignore warning and use null jvmId
                     if (!(AbstractAuthenticatedRequestHandler.isJmxAuthFailure(e)
                             || AbstractAuthenticatedRequestHandler.isJmxSslFailure(e))) {
@@ -323,7 +375,6 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                         continue;
                     }
                     logger.info("Update node [{}] with null jvmId", child.getName());
-                    targetsToUpdate.putIfAbsent((TargetNode) child, id);
                 }
                 child = new TargetNode(child.getNodeType(), ref, child.getLabels());
                 modifiedChildren.add(child);

--- a/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
+++ b/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
@@ -45,11 +45,13 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ForkJoinPool;
+import java.util.function.Predicate;
 
 import javax.script.ScriptException;
 
@@ -149,48 +151,40 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
         this.targetRetryTimeId =
                 getVertx()
                         .setPeriodic(
+                                // TODO make this configurable, use an exponential backoff, have a
+                                // maximum retry policy, etc.
                                 15_000,
                                 i -> {
                                     ForkJoinPool.commonPool()
                                             .execute(
-                                                    () -> {
-                                                        Map<TargetNode, UUID> copy =
-                                                                new HashMap<>(
-                                                                        nonConnectableTargets);
-                                                        for (var entry : copy.entrySet()) {
-                                                            TargetNode targetNode = entry.getKey();
-                                                            try {
-                                                                ServiceRef sr =
-                                                                        jvmIdHelper
-                                                                                .get()
-                                                                                .resolveId(
-                                                                                        targetNode
-                                                                                                .getTarget());
-                                                                if (StringUtils.isBlank(
-                                                                        sr.getJvmId())) {
-                                                                    continue;
-                                                                }
-                                                                nonConnectableTargets.remove(
-                                                                        targetNode);
-                                                                UUID id = entry.getValue();
-                                                                PluginInfo plugin =
-                                                                        getById(id).orElseThrow();
-                                                                EnvironmentNode original =
-                                                                        gson.fromJson(
-                                                                                plugin.getSubtree(),
-                                                                                EnvironmentNode
-                                                                                        .class);
-                                                                update(id, original.getChildren());
-                                                            } catch (JvmIdGetException e) {
-                                                                logger.info(
-                                                                        "Retain null jvmId for node"
-                                                                                + " [{}]",
-                                                                        targetNode);
-                                                            } catch (JsonSyntaxException e) {
-                                                                throw new RuntimeException(e);
-                                                            }
-                                                        }
-                                                    });
+                                                    () ->
+                                                            testNonConnectedTargets(
+                                                                    entry -> {
+                                                                        TargetNode targetNode =
+                                                                                entry.getKey();
+                                                                        try {
+                                                                            String id =
+                                                                                    jvmIdHelper
+                                                                                            .get()
+                                                                                            .resolveId(
+                                                                                                    targetNode
+                                                                                                            .getTarget())
+                                                                                            .getJvmId();
+                                                                            return StringUtils
+                                                                                    .isNotBlank(id);
+                                                                        } catch (
+                                                                                JvmIdGetException
+                                                                                        e) {
+                                                                            logger.info(
+                                                                                    """
+                                                                                    Retain null jvmId for node [{}]
+                                                                                    """,
+                                                                                    targetNode
+                                                                                            .getName());
+                                                                            logger.info(e);
+                                                                            return false;
+                                                                        }
+                                                                    }));
                                 });
         this.credentialsManager
                 .get()
@@ -198,28 +192,19 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                         event -> {
                             switch (event.getEventType()) {
                                 case ADDED:
-                                    Map<TargetNode, UUID> copy =
-                                            new HashMap<>(nonConnectableTargets);
-                                    for (var entry : copy.entrySet()) {
-                                        try {
-                                            if (matchExpressionEvaluator
-                                                    .get()
-                                                    .applies(
-                                                            event.getPayload(),
-                                                            entry.getKey().getTarget())) {
-                                                nonConnectableTargets.remove(entry.getKey());
-                                                UUID id = entry.getValue();
-                                                PluginInfo plugin = getById(id).orElseThrow();
-                                                EnvironmentNode original =
-                                                        gson.fromJson(
-                                                                plugin.getSubtree(),
-                                                                EnvironmentNode.class);
-                                                update(id, original.getChildren());
-                                            }
-                                        } catch (JsonSyntaxException | ScriptException e) {
-                                            throw new RuntimeException(e);
-                                        }
-                                    }
+                                    testNonConnectedTargets(
+                                            entry -> {
+                                                try {
+                                                    return matchExpressionEvaluator
+                                                            .get()
+                                                            .applies(
+                                                                    event.getPayload(),
+                                                                    entry.getKey().getTarget());
+                                                } catch (ScriptException e) {
+                                                    logger.error(e);
+                                                    return false;
+                                                }
+                                            });
                                     break;
                                 case REMOVED:
                                     break;
@@ -228,6 +213,24 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                                             event.getEventType().toString());
                             }
                         });
+    }
+
+    private void testNonConnectedTargets(Predicate<Entry<TargetNode, UUID>> predicate) {
+        Map<TargetNode, UUID> copy = new HashMap<>(nonConnectableTargets);
+        for (var entry : copy.entrySet()) {
+            try {
+                if (predicate.test(entry)) {
+                    nonConnectableTargets.remove(entry.getKey());
+                    UUID id = entry.getValue();
+                    PluginInfo plugin = getById(id).orElseThrow();
+                    EnvironmentNode original =
+                            gson.fromJson(plugin.getSubtree(), EnvironmentNode.class);
+                    update(id, original.getChildren());
+                }
+            } catch (JsonSyntaxException e) {
+                throw new RuntimeException(e);
+            }
+        }
     }
 
     @Override

--- a/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
+++ b/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
@@ -58,7 +58,6 @@ import io.cryostat.configuration.CredentialsManager;
 import io.cryostat.configuration.StoredCredentials;
 import io.cryostat.core.log.Logger;
 import io.cryostat.core.net.discovery.JvmDiscoveryClient.EventKind;
-import io.cryostat.net.web.http.AbstractAuthenticatedRequestHandler;
 import io.cryostat.platform.ServiceRef;
 import io.cryostat.platform.ServiceRef.AnnotationKey;
 import io.cryostat.platform.discovery.AbstractNode;
@@ -366,17 +365,12 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                 ServiceRef ref = ((TargetNode) child).getTarget();
                 try {
                     ref = jvmIdHelper.get().resolveId(ref);
+                    child = new TargetNode(child.getNodeType(), ref, child.getLabels());
                 } catch (Exception e) {
-                    nonConnectableTargets.putIfAbsent((TargetNode) child, id);
-                    // if Exception is of SSL or JMX Auth, ignore warning and use null jvmId
-                    if (!(AbstractAuthenticatedRequestHandler.isJmxAuthFailure(e)
-                            || AbstractAuthenticatedRequestHandler.isJmxSslFailure(e))) {
-                        logger.info("Ignoring target node [{}]", child.getName());
-                        continue;
-                    }
                     logger.info("Update node [{}] with null jvmId", child.getName());
+                    logger.info(e);
+                    nonConnectableTargets.putIfAbsent((TargetNode) child, id);
                 }
-                child = new TargetNode(child.getNodeType(), ref, child.getLabels());
                 modifiedChildren.add(child);
             } else if (child instanceof EnvironmentNode) {
                 modifiedChildren.add(

--- a/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
+++ b/src/main/java/io/cryostat/discovery/DiscoveryStorage.java
@@ -155,36 +155,25 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
                                 // maximum retry policy, etc.
                                 15_000,
                                 i -> {
-                                    ForkJoinPool.commonPool()
-                                            .execute(
-                                                    () ->
-                                                            testNonConnectedTargets(
-                                                                    entry -> {
-                                                                        TargetNode targetNode =
-                                                                                entry.getKey();
-                                                                        try {
-                                                                            String id =
-                                                                                    jvmIdHelper
-                                                                                            .get()
-                                                                                            .resolveId(
-                                                                                                    targetNode
-                                                                                                            .getTarget())
-                                                                                            .getJvmId();
-                                                                            return StringUtils
-                                                                                    .isNotBlank(id);
-                                                                        } catch (
-                                                                                JvmIdGetException
-                                                                                        e) {
-                                                                            logger.info(
-                                                                                    """
-                                                                                    Retain null jvmId for node [{}]
-                                                                                    """,
-                                                                                    targetNode
-                                                                                            .getName());
-                                                                            logger.info(e);
-                                                                            return false;
-                                                                        }
-                                                                    }));
+                                    testNonConnectedTargets(
+                                            entry -> {
+                                                TargetNode targetNode = entry.getKey();
+                                                try {
+                                                    String id =
+                                                            jvmIdHelper
+                                                                    .get()
+                                                                    .resolveId(
+                                                                            targetNode.getTarget())
+                                                                    .getJvmId();
+                                                    return StringUtils.isNotBlank(id);
+                                                } catch (JvmIdGetException e) {
+                                                    logger.info(
+                                                            "Retain null jvmId for node [{}]",
+                                                            targetNode.getName());
+                                                    logger.info(e);
+                                                    return false;
+                                                }
+                                            });
                                 });
         this.credentialsManager
                 .get()
@@ -216,21 +205,26 @@ public class DiscoveryStorage extends AbstractPlatformClientVerticle {
     }
 
     private void testNonConnectedTargets(Predicate<Entry<TargetNode, UUID>> predicate) {
-        Map<TargetNode, UUID> copy = new HashMap<>(nonConnectableTargets);
-        for (var entry : copy.entrySet()) {
-            try {
-                if (predicate.test(entry)) {
-                    nonConnectableTargets.remove(entry.getKey());
-                    UUID id = entry.getValue();
-                    PluginInfo plugin = getById(id).orElseThrow();
-                    EnvironmentNode original =
-                            gson.fromJson(plugin.getSubtree(), EnvironmentNode.class);
-                    update(id, original.getChildren());
-                }
-            } catch (JsonSyntaxException e) {
-                throw new RuntimeException(e);
-            }
-        }
+        ForkJoinPool.commonPool()
+                .execute(
+                        () -> {
+                            Map<TargetNode, UUID> copy = new HashMap<>(nonConnectableTargets);
+                            for (var entry : copy.entrySet()) {
+                                try {
+                                    if (predicate.test(entry)) {
+                                        nonConnectableTargets.remove(entry.getKey());
+                                        UUID id = entry.getValue();
+                                        PluginInfo plugin = getById(id).orElseThrow();
+                                        EnvironmentNode original =
+                                                gson.fromJson(
+                                                        plugin.getSubtree(), EnvironmentNode.class);
+                                        update(id, original.getChildren());
+                                    }
+                                } catch (JsonSyntaxException e) {
+                                    throw new RuntimeException(e);
+                                }
+                            }
+                        });
     }
 
     @Override

--- a/src/main/java/io/cryostat/recordings/JvmIdHelper.java
+++ b/src/main/java/io/cryostat/recordings/JvmIdHelper.java
@@ -119,7 +119,6 @@ public class JvmIdHelper extends AbstractEventEmitter<JvmIdHelper.IdEvent, Strin
     }
 
     private boolean observe(ServiceRef sr) {
-        logger.info("Observing new target: {}", sr);
         if (StringUtils.isBlank(sr.getJvmId())) {
             return false;
         }
@@ -133,6 +132,7 @@ public class JvmIdHelper extends AbstractEventEmitter<JvmIdHelper.IdEvent, Strin
         if (observe(sr)) {
             return sr;
         }
+        logger.info("Observing new target: {}", sr);
         URI serviceUri = sr.getServiceUri();
         String uriStr = serviceUri.toString();
         try {

--- a/src/test/java/io/cryostat/discovery/DiscoveryStorageTest.java
+++ b/src/test/java/io/cryostat/discovery/DiscoveryStorageTest.java
@@ -47,7 +47,6 @@ import java.util.Optional;
 import java.util.Set;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutionException;
 
 import javax.inject.Singleton;
 
@@ -68,7 +67,6 @@ import io.cryostat.platform.internal.CustomTargetPlatformClient;
 import io.cryostat.platform.internal.DefaultPlatformClient;
 import io.cryostat.platform.internal.KubeApiPlatformClient;
 import io.cryostat.recordings.JvmIdHelper;
-import io.cryostat.recordings.JvmIdHelper.JvmIdGetException;
 import io.cryostat.rules.MatchExpressionEvaluator;
 
 import com.google.gson.Gson;
@@ -915,86 +913,6 @@ class DiscoveryStorageTest {
                             new TargetDiscoveryEvent(EventKind.FOUND, updatedServiceRef2),
                             new TargetDiscoveryEvent(EventKind.FOUND, updatedServiceRef3),
                             new TargetDiscoveryEvent(EventKind.MODIFIED, updatedServiceRef4)));
-        }
-
-        @Test
-        void testIgnoreNodesIfNonSSLAuthExceptions() throws Exception {
-            UUID id = UUID.randomUUID();
-
-            ServiceRef serviceRef1 =
-                    new ServiceRef(
-                            null,
-                            URI.create("service:jmx:rmi:///jndi/rmi://localhost:1/jmxrmi"),
-                            "serviceRef1");
-            ServiceRef serviceRef2 =
-                    new ServiceRef(
-                            null,
-                            URI.create("service:jmx:rmi:///jndi/rmi://localhost:2/jmxrmi"),
-                            "serviceRef2");
-            TargetNode target1 = new TargetNode(BaseNodeType.JVM, serviceRef1);
-            TargetNode target2 = new TargetNode(BaseNodeType.JVM, serviceRef2);
-
-            EnvironmentNode realm1 =
-                    new EnvironmentNode("next", BaseNodeType.REALM, Map.of(), Set.of(target1));
-            EnvironmentNode realm2 =
-                    new EnvironmentNode(
-                            "next", BaseNodeType.REALM, Map.of(), Set.of(target1, target2));
-
-            PluginInfo prevPlugin =
-                    new PluginInfo(
-                            "test-realm", URI.create("http://example.com"), gson.toJson(realm1));
-
-            Mockito.when(dao.get(Mockito.eq(id))).thenReturn(Optional.of(prevPlugin));
-            Mockito.when(dao.update(Mockito.any(UUID.class), Mockito.any(Collection.class)))
-                    .thenAnswer(
-                            new Answer<PluginInfo>() {
-                                @Override
-                                public PluginInfo answer(InvocationOnMock invocation)
-                                        throws Throwable {
-                                    List<AbstractNode> subtree = invocation.getArgument(1);
-                                    EnvironmentNode next =
-                                            new EnvironmentNode(
-                                                    "next", BaseNodeType.REALM, Map.of(), subtree);
-                                    return new PluginInfo(
-                                            "test-realm",
-                                            URI.create("http://example.com"),
-                                            gson.toJson(next));
-                                }
-                            });
-            JvmIdGetException jige = Mockito.mock(JvmIdGetException.class);
-            ExecutionException ex = Mockito.mock(ExecutionException.class);
-            Mockito.when(jige.getCause()).thenReturn(ex);
-            Mockito.when(ex.getCause()).thenReturn(new SecurityException("test"));
-
-            Mockito.when(jvmIdHelper.resolveId(Mockito.any(ServiceRef.class)))
-                    .thenThrow(jige)
-                    .thenReturn(
-                            new ServiceRef(
-                                    "serviceRef2", serviceRef2.getServiceUri(), "serviceRef2"));
-
-            var updatedSubtree = storage.update(id, List.of(realm2));
-            MatcherAssert.assertThat(updatedSubtree, Matchers.notNullValue());
-            MatcherAssert.assertThat(updatedSubtree, Matchers.hasSize(1));
-            for (AbstractNode node : updatedSubtree) {
-                MatcherAssert.assertThat(node, Matchers.instanceOf(EnvironmentNode.class));
-                EnvironmentNode env = (EnvironmentNode) node;
-                MatcherAssert.assertThat(env.getChildren(), Matchers.hasSize(2));
-                for (AbstractNode nested : env.getChildren()) {
-                    if (nested instanceof TargetNode) {
-                        TargetNode target = (TargetNode) nested;
-                        MatcherAssert.assertThat(target, Matchers.instanceOf(TargetNode.class));
-                        if (target.getTarget().getAlias().get().equals("serviceRef1")) {
-                            MatcherAssert.assertThat(
-                                    target.getTarget().getJvmId(), Matchers.nullValue());
-                        } else if (target.getTarget().getAlias().get().equals("serviceRef2")) {
-                            MatcherAssert.assertThat(
-                                    target.getTarget().getJvmId(), Matchers.equalTo("serviceRef2"));
-                        } else {
-                            throw new IllegalStateException("Unexpected alias");
-                        }
-                    }
-                }
-            }
         }
     }
 }

--- a/src/test/java/itest/bases/ExternalTargetsTest.java
+++ b/src/test/java/itest/bases/ExternalTargetsTest.java
@@ -45,7 +45,7 @@ import org.junit.jupiter.api.AfterAll;
 
 public abstract class ExternalTargetsTest extends StandardSelfTest {
 
-    protected static final String FIB_DEMO_IMAGESPEC = "quay.io/andrewazores/vertx-fib-demo:0.8.0";
+    protected static final String FIB_DEMO_IMAGESPEC = "quay.io/andrewazores/vertx-fib-demo:0.13.0";
 
     static final int DISCOVERY_POLL_PERIOD_MS =
             Integer.parseInt(System.getProperty("cryostat.itest.jdp.poll.period", "2500"));

